inference-engine/src/transformations/src/transformations/common_optimizations/simplify_shape_of_sub_graph.cpp

```diff
@@ -121,16 +122,85 @@ ngraph::pass::GatherNopElimination::GatherNopElimination() {
     this->register_matcher(m, callback);
 }
 
+NGRAPH_RTTI_DEFINITION(ngraph::pass::SimplifyGatherShapeOf, "SimplifyGatherShapeOf", 0);
+
+ngraph::pass::SimplifyGatherShapeOf::SimplifyGatherShapeOf() {
+    MATCHER_SCOPE(SimplifyGatherShapeOf);
+    const auto gather_pattern = ngraph::pattern::wrap_type<op::util::GatherBase>();
+    const auto shape_of_pattern = ngraph::pattern::wrap_type<opset2::ShapeOf, opset3::ShapeOf>({gather_pattern});
+
+    ngraph::matcher_pass_callback callback = [](pattern::Matcher& m) {
+        auto node = m.get_match_root();
+        auto gather = as_type_ptr<opset3::Gather>(node->input_value(0).get_node_shared_ptr());
+        if (!gather) {
+            return false;
+        }
```

Comment:
we match gather base but require opset3 version.

Reply:
Discussed offline that this transformation needs to be updated to support new Gather versions (opset7, opset8). The ticket in transformations backlog will be created.

Reply:
Ticket: XXX-61783